### ISSUE TITLE
Inifinite loop when wiki directory contains cyclical links

### ISSUE THREAD

**Issue by mnzaki** [vimwiki/vimwiki#1302]
If the wiki directory contains a link pointing to the parent directory then vim wiki traverses forever / hangs forever while loading a new markdown file

### .vimrc
```
" VimWiki
let wiki = {
\ 'path': '~/pkb',
\ 'maxhi': 1,
\ 'path_html': '~/pkb.html/',
\ 'nested_syntaxes': {
  \ 'python': 'python', 'c++': 'cpp', 'c': 'c',
  \ 'ruby': 'ruby', 'perl': 'perl', 'sh': 'sh'
\ },
\ 'syntax': 'markdown', 'ext': '.md', 'auto_tags': 1 }

let g:vimwiki_list = [wiki]
```
### `:VimwikiShowVersion`
```
Version: 2022.12.02
Os: Linux
Vim: 900
Branch: master
Revision: fea8bee
Date: 2022-12-04 21:25:52 -0700
```

**Comment by brennen** [comment]
I would gently suggest not doing that, but thanks for the report. :)

Without digging into code, I'm not sure why it's doing that at all,but I'm going to guess it's fixable. Seems like there might be something here similar to #1267.

**Comment by mnzaki** [comment]
I cannot reproduce it now either. Maybe fixed since `Version: 2022.12.02`?
